File: app/src/main/java/com/billiamram/hexagonalscaffold/AndroidAdaptersModule.java
package com.billiamram.hexagonalscaffold;

import android.content.SharedPreferences;

import com.billiamram.main.ExampleModelRepository;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidAdaptersModule {

    @Singleton
    @Provides
    ExampleModelRepository provideItemRepository(SharedPreferences sharedPreferences,
                                                 Gson gson) {
        return new AndroidExampleModelRepository(sharedPreferences, gson);
    }

}

File: app/src/main/java/com/billiamram/hexagonalscaffold/AndroidAdaptersModule.kt
package com.billiamram.hexagonalscaffold

import android.content.SharedPreferences
import com.billiamram.main.ExampleModelRepository
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AndroidAdaptersModule {

    @Singleton
    @Provides
    internal fun provideItemRepository(sharedPreferences: SharedPreferences,
                                       gson: Gson): ExampleModelRepository {
        return AndroidExampleModelRepository(sharedPreferences, gson)
    }

    @Singleton
    @Provides
    fun provideGson(): Gson {
        return GsonBuilder()
                .create()
    }

}
